### PR TITLE
 DR nav fixes + tests for AvatarStack, PartnerSlot and CommunityBar

### PR DESCRIPTION
**Pull request by xkxd**
@Wikia/iwing

### CODE REVIEW

File: source/components/GlobalNavigation/components/Search/Search.spec.js [hunk 1]
```diff
@@ -256,556 +256,567 @@ describe('onKeyDown', () => {
...
-            const stateMock = { suggestions: suggestionsMock, suggestionId: suggestionIdMock };
-            const onSearchSuggestionChosenMock = jest.fn();
-            const onSearchCloseMock = jest.fn();
-            const trackMock = jest.fn();
-            const wrapper = renderShallowComponent({
-                onSearchSuggestionChosen: onSearchSuggestionChosenMock,
-                track: trackMock,
-            }).setState(stateMock);
-            const instance = wrapper.instance();
+describe('onSearchSuggestionClick', () => {
```
Comment (xkxd) [inline source/components/GlobalNavigation/components/Search/Search.spec.js]: this code was only wrongly nested.

File: source/components/GlobalNavigation/components/Notifications/MarkAllButton.js [hunk 2]
```diff
@@ -20,9 +20,10 @@ const MarkAllButton = ({ track, header }) => {
             {
                 header && <p className="wds-notifications__header">{header}</p>
             }
-            <button onClick={onClick} className="wds-notifications__mark-all-as-read">
+            {/* eslint-disable-next-line jsx-a11y/anchor-is-valid */}
+            <a onClick={onClick} className="wds-notifications__mark-all-as-read">
```
Comment (jsutterfield) [inline source/components/GlobalNavigation/components/Notifications/MarkAllButton.js]: just curious, why change this from a button to an `a` tag? Are there styles we're re-using or something?

File: source/components/GlobalNavigation/components/Search/Search.js [hunk 3]
```diff
@@ -354,37 +361,40 @@ class Search extends React.Component {
                     onClick={this.onSearchClose}
                     text
                 >
-                    <Icon name="close" className="wds-global-navigation__search-close-icon" tiny />
+                    <Icon name="close-tiny" className="wds-global-navigation__search-close-icon" tiny />
```
Comment (jsutterfield) [inline source/components/GlobalNavigation/components/Search/Search.js]: `Icon` is deprecated, right? Maybe swap this with `@wikia/react-common/icons/IconCloseTiny`?

File: package.json [hunk 4]
```diff
@@ -45,7 +45,8 @@
     "react-dom": "^16.8.3",
     "react-i18next": "^10.0.0",
     "react-redux": "^5.0.7",
-    "react-select": "^2.1.1"
+    "react-select": "^2.1.1",
+    "uuid": "^3.3.2"
```
Comment (jsutterfield) [inline package.json]: I don't see this dependency being used in this PR. Can we remove it? Or is it used elsewhere?

File: source/components/GlobalNavigation/components/Notifications/MarkAllButton.js (hunk 2 above)
Comment (xkxd) [reply source/components/GlobalNavigation/components/Notifications/MarkAllButton.js]: exactly. it's defined as an `a` tag in original DS and the styling was done in order to cover that type

File: package.json (hunk 4 above)
Comment (xkxd) [reply package.json]: it's used in Search.js https://github.com/Wikia/react-common/pull/22/files#diff-29c98872cca9a2d080301634fa42a45bR6 but apparently disappeared during previous PRs

File: source/components/GlobalNavigation/components/Search/Search.js (hunk 3 above)
Comment (xkxd) [reply source/components/GlobalNavigation/components/Search/Search.js]: at the moment of creation, there were issues with `IconXXX` components which are now resolved. I'll go through the whole GN codebase and update it in the last PR

**Pull request merged**